test/run_tests.sh

```diff
@@ -3,7 +3,7 @@
 docker build -f test/Dockerfile -t kudo-test .
 if [ $? -eq 0 ]
 then
-   docker run kudo-test
+   docker run -it -m 4g --rm kudo-test
```

Comment:
4 gigs is arbitrary. I set it to 1 gig and the tests hung.

Reply:
I did some experimentation, and the tests seem to peak at about 1.5 gb RAM usage while running some of the unit tests. Dropping it to 2gb.

Reply:
When running this all night, it ended up hanging with 2gb :( I'm going to bump it up to 4gb and give us more head room.